code/obj/machinery/dispenser.dm

```diff
@@ -25,32 +34,20 @@
 				return
 		if(3.0)
 			if (prob(25))
-				while(src.o2tanks > 0)
-					new /obj/item/tank/oxygen( src.loc )
-					src.o2tanks--
-				while(src.pltanks > 0)
-					new /obj/item/tank/plasma( src.loc )
-					src.pltanks--
+				pop_o2()
```

Comment:
The old behavior was dumping all the tanks on the floor- doesn't this just dump one of each
i.e. should probably stick these in the loops instead of removing the loops

Reply:
Oh I must have misinterpreted those as if statements